File: packages/domain/src/ports/IClientRepository.ts
import type { Client } from '../entities/client.js';
import type { ClientProps } from '../entities/client.types.js';

export interface IClientRepository {
  findAll(): Promise<Client[]>;
  findById(id: string): Promise<Client | null>;
  create(props: ClientProps): Promise<Client>;
  update(id: string, props: Partial<Pick<ClientProps, 'name' | 'email'>>): Promise<Client>;
  delete(id: string): Promise<void>;
  exists(id: string): Promise<boolean>;
}

File: backend/dotnet/src/Monolegal.Domain/Ports/IClientRepository.cs
using Monolegal.Domain.Entities;

namespace Monolegal.Domain.Ports;

public interface IClientRepository
{
    Task<IReadOnlyList<Client>> FindAllAsync(CancellationToken cancellationToken = default);
    Task<Client?> FindByIdAsync(string id, CancellationToken cancellationToken = default);
    Task<Client> CreateAsync(ClientProps props, CancellationToken cancellationToken = default);
    Task<Client> UpdateAsync(string id, string? name, string? email, CancellationToken cancellationToken = default);
    Task DeleteAsync(string id, CancellationToken cancellationToken = default);
    Task<bool> ExistsAsync(string id, CancellationToken cancellationToken = default);
}
